tests/test_integration.py

```diff
@@ -1338,3 +1340,73 @@ def test_issues_539_abi3_resolution():
     res.assert_success()
 
     subprocess.check_call([cryptography_pex, '-c', 'import cryptography'])
+
+
+def assert_reproducible_build(args):
+  with temporary_dir() as td:
+    pex1 = os.path.join(td, '1.pex')
+    pex2 = os.path.join(td, '2.pex')
+    # Note that we change the `PYTHONHASHSEED` to ensure that there are no issues resulting
+    # from the random seed, such as data structures, as Tox sets this value by default. See
+    # https://tox.readthedocs.io/en/latest/example/basic.html#special-handling-of-pythonhashseed.
+    run_pex_command(args + ['-o', pex1], env=make_env(PYTHONHASHSEED=111))
+    run_pex_command(args + ['-o', pex2], env=make_env(PYTHONHASHSEED=22222))
+    # First explode the PEXes to compare file-by-file for easier debugging.
+    with ZipFile(pex1) as zf1, ZipFile(pex2) as zf2:
+      unzipped1 = os.path.join(td, "pex1")
+      unzipped2 = os.path.join(td, "pex2")
+      zf1.extractall(path=unzipped1)
+      zf2.extractall(path=unzipped2)
+      for member1, member2 in zip(sorted(zf1.namelist()), sorted(zf2.namelist())):
+        assert filecmp.cmp(
+          os.path.join(unzipped1, member1),
+          os.path.join(unzipped2, member2),
+          shallow=False
+        )
+    # Then compare the original .pex files. This is the assertion we truly care about.
+    assert filecmp.cmp(pex1, pex2, shallow=False)
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build_no_args():
+  assert_reproducible_build([])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build_bdist_requirements():
+  # We test both a pure Python wheel (six) and a platform-specific wheel (cryptography).
+  assert_reproducible_build(['six==1.12.0', 'cryptography==2.6.1'])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build_sdist_requirements():
+  assert_reproducible_build(['pycparser==2.19', '--no-build'])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build_m_flag():
+  assert_reproducible_build(['-m', 'pydoc'])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build_c_flag():
```

Comment:
@jsirois does this test look right? Do I need `zip_safe=True`?

The PEXes get built without error and only fail because they aren't equal to each other, so I think it's working correctly.

Reply:
The `zip_safe` characterisitc is completely orthoganal to `-c`. If `-c` is a `console_scripts` entry point, thats found via `sys.path` and if its a script, that's found via resource files (so also via `sys.path`). So this looks fine, but it only tests 1/2 of -c. I'd move on though and circle back to test the other half later.